bolt/lib/Core/MCPlusBuilder.cpp

```diff
@@ -269,15 +269,17 @@ bool MCPlusBuilder::clearOffset(MCInst &Inst) {
 }
 
 std::optional<MCSymbol *> MCPlusBuilder::getLabel(const MCInst &Inst) const {
-  if (auto Label = tryGetAnnotationAs<MCSymbol *>(Inst, MCAnnotation::kLabel))
-    return *Label;
-  return std::nullopt;
+  std::optional<int64_t> Label =
+      getAnnotationOpValue(Inst, MCAnnotation::kLabel);
+  if (!Label)
+    return std::nullopt;
+
+  return reinterpret_cast<MCSymbol *>(*Label);
 }
 
-bool MCPlusBuilder::setLabel(MCInst &Inst, MCSymbol *Label,
-                             AllocatorIdTy AllocatorId) {
-  getOrCreateAnnotationAs<MCSymbol *>(Inst, MCAnnotation::kLabel, AllocatorId) =
-      Label;
+bool MCPlusBuilder::setLabel(MCInst &Inst, MCSymbol *Label) {
```

Comment:
Why was `AllocatorId` removed? Don't we still need to pass it to `setAnnotationOpValue`?

Reply:
Yes, you are absolutely right. I have a diff that refactors annotations in a way that `AllocatorId` will not be needed for `setLabel()`, but until then you might need `AllocatorId` depending on the use case.

Reply:
Yes, I guess so. #70412 has more details. I also plan to use annotations more often, hence I wanted to reduce the associated overhead.